engine/consensus/matching/core_test.go

```diff
@@ -178,14 +179,13 @@ func (ms *MatchingSuite) TestOnReceiptPendingResult() {
 	// Expect the receipt to be added to mempool
 	ms.ReceiptsPL.On("AddReceipt", receipt, ms.UnfinalizedBlock.Header).Return(true, nil).Once()
 
-	err := ms.matching.OnReceipt(receipt.ExecutorID, receipt)
+	_, err := ms.matching.onCurrentReceipt(receipt)
 	ms.Require().NoError(err, "should not error for different receipt for already pending result")
 	ms.ReceiptsPL.AssertExpectations(ms.T())
-	ms.ResultsPL.AssertExpectations(ms.T())
```

Comment:
Similar answer here to https://github.com/onflow/flow-go/pull/387/files#r574283257

The tests should be written as you said, the mempool should have a receipt stored there. However, the mock makes it harder to check. 

I think we could address in a separate issue to refactor the test with real mempool, instead of mocks

Reply:
the problem is the addition of these lines: https://github.com/onflow/flow-go/blob/93c34ee74265d90163dfbe451997fc8491b17399/utils/unittest/chain_suite.go#L399-L405

this covers most calls and therefore, your more specific matcher is shortcutted https://github.com/onflow/flow-go/blob/93c34ee74265d90163dfbe451997fc8491b17399/engine/consensus/matching/core_test.go#L172-L177   
essentially, you are weakening the test !

Reply:
updated proposed in https://github.com/onflow/flow-go/pull/403